src/main/java/edu/cornell/kfs/module/cam/document/service/impl/CuAssetServiceImpl.java

```diff
@@ -53,7 +59,17 @@ public Asset updateAssetInventory(String capitalAssetNumber, String conditionCod
         Timestamp currentTimestamp = dateTimeService.getCurrentTimestamp();
         asset.setLastInventoryDate(currentTimestamp);
         asset.setLastUpdatedTimestamp(currentTimestamp);
-        return businessObjectService.save(asset);
+        asset = businessObjectService.save(asset);
+
+        AssetExtension assetExtension = businessObjectService.findBySinglePrimaryKey(AssetExtension.class, capitalAssetNumber);
```

Comment:
you should be able to do 
AssetExtension assetExtension - (AssetExtension) asset.getExtension()

you'll want to check for null, and create a new instance of AssetExtension as needed.  If you need to create a new AssetExtension object, you'll want to call asset.setExtension().

doing a business object save of asset should save both the asset and asset extension.